### PR TITLE
CorreatedJoin prunning must not add SelectSymbol to the input plan outputs

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Fixes https://github.com/crate/crate/issues/17016

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
Problem: Eval operator's prunning logic doesn't have a check that outputsToKeep must be a part of the original output. Sometimes it can lead to a creation of an `Eval` operator with unexpected outputs and such `addEvalProjection` failing due to unresolved `InputColumn`

Int the reported scenario SelectSymbol was added in  
CorrelatedJoin (input = Eval) -> input.pruneOutputsExcept (SelectSymbols) -> Eval (..., ..., SelectSymbol).

Eval popped up due to reordering rule being applied. 

If we don't have state, rule is not applied, we don't have Eval, we have `CorrelatedJoin` with input plan = `HashJoin` which doesn't ad SelectSymbol to its outputs when doing prunnig.

**Comment by mfussenegger** [comment]
> Sometimes it can lead to a creation of an Eval operator with unexpected outputs

> In the reported scenario SelectSymbol was added in
CorrelatedJoin (input = Eval) -> input.pruneOutputsExcept (SelectSymbols) -> Eval (..., ..., SelectSymbol).


Why is that being added if we're going to prune it anyway?

**Comment by BaurzhanSakhariev** [comment]
>Why is that being added if we're going to prune it anyway?

I think it's a misusage of the outputs to keep (which we need to create a new CorrelatedJoin) but looks like for prunning we can used a filtered version - exclude symbols that are correlated? ie 
` LogicalPlan newInputPlan = inputPlan.pruneOutputsExcept(toCollect.filter (s -s.isCorrelated() == false));`

**Comment by BaurzhanSakhariev** [comment]
Hi @mfussenegger, I pushed another approach, can you please take a look?

File: docs/appendices/release-notes/5.9.4.rst [hunk 1]
```diff
@@ -47,6 +47,10 @@ See the :ref:`version_5.9.0` release notes for a full list of changes in the
 Fixes
 =====
 
+- Fixed an issue leading to an error when running ``ANALYZE`` followed by a
+  correlated subquery with the sub-query being an ``INNER JOIN`` and a table on
```
Comment (mfussenegger) [inline docs/appendices/release-notes/5.9.4.rst:51]:
```suggestion
  correlated subquery with the Subquery being an ``INNER JOIN`` and a table on
```

File: server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java [hunk 2]
```diff
@@ -707,4 +708,50 @@ Eval[CASE WHEN (NOT (regexp_matches(name, '^a') = [])) THEN 'found' WHEN (text L
...
+                FROM t1 g
+                WHERE g.i = 1
+            )
+            SELECT (
+                    SELECT true
+                    FROM t1 s
+                    WHERE c.x = s.x
+                   )
+            FROM cte_1 c
+            JOIN t2 p ON c.x = p.y"""
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java:735]: Are the where clauses relevant for the scenario? If not I'd remove them.

**Review by mfussenegger — APPROVED** [review]

**Comment by BaurzhanSakhariev** [comment]
Updated commit message

**Pull request merged**